src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -63,8 +63,18 @@ protected function createController($controller)
                 list($service, $method) = explode(':', $controller, 2);
 
                 return array($this->container->get($service), $method);
-            } elseif ($this->container->has($controller) && method_exists($service = $this->container->get($controller), '__invoke')) {
-                return $service;
+            } elseif ($this->container->has($controller)) {
+                $controllerInstance = $this->container->get($controller);
```

Comment:
$controllerService instead of $controllerInstance? or just $service as it was before